pkg/ruler/worker.go

```diff
@@ -1,30 +1,30 @@
 package ruler
 
 import (
+	"context"
+	native_ctx "context"
 	"time"
 
 	"github.com/cortexproject/cortex/pkg/util"
 	"github.com/go-kit/kit/log/level"
+	opentracing "github.com/opentracing/opentracing-go"
 	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/weaveworks/common/instrument"
+	"github.com/weaveworks/common/user"
 )
 
 var (
-	blockedWorkers = prometheus.NewGauge(prometheus.GaugeOpts{
+	blockedWorkers = promauto.NewGauge(prometheus.GaugeOpts{
 		Namespace: "cortex",
 		Name:      "blocked_workers",
 		Help:      "How many workers are waiting on an item to be ready.",
 	})
-	workerIdleTime = prometheus.NewCounter(prometheus.CounterOpts{
+	workerIdleTime = promauto.NewCounter(prometheus.CounterOpts{
 		Namespace: "cortex",
 		Name:      "worker_idle_seconds_total",
 		Help:      "How long workers have spent waiting for work.",
 	})
-	evalLatency = prometheus.NewHistogram(prometheus.HistogramOpts{
```

Comment:
This is moved to the scheduler with different buckets and an additional metric for missed iterations